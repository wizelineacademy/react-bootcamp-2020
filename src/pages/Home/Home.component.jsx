import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import {
  Container,
  Title,
  VideoContainer,
  Videos,
  ThumbnailContainer,
  InfoContainer,
  VideoName,
  ChannelName,
  Description,
} from './Home.styles';
import { useAuth } from '../../providers/Auth';
import { toggleSideNav } from '../../components/SideNav';
import thumbnail from '../../images/thumb.jpg';
import { storage } from '../../utils/storage';

function Main() {
  const { state, setState } = useAuth();
  const history = useHistory();
  const { sidenav, videos } = state;
  const currentPath = history.location.pathname;
  const title = currentPath === '/favorites' ? 'Favorites' : 'Trending';
  const storagedVideos = storage.get('localVideoStoraged');
  const videoList =
    (currentPath === '/favorites'
      ? videos.filter((video) => video.isFavorite) ||
        storagedVideos.video.filter((video) => video.isFavorite)
      : videos) || new Array(16).fill(null);

  let swipeStart;
  let swipeEnd;

  function handleSwipe() {
    if (swipeEnd > swipeStart + 100 && !sidenav) {
      toggleSideNav({ state, setState });
    } else if (swipeStart > swipeEnd + 100 && sidenav) {
      toggleSideNav({ state, setState });
    }
  }

  function touchStart(event) {
    const lastTouchIndex = event.changedTouches.length - 1;
    swipeStart = event.changedTouches[lastTouchIndex].screenX;
  }

  function touchEnd(event) {
    const lastTouchIndex = event.changedTouches.length - 1;
    swipeEnd = event.changedTouches[lastTouchIndex].screenX;
    handleSwipe();
  }

  function handleOnSelectVideo(videoId) {
    setState({
      ...state,
      currentVideoId: videoId,
    });

    if (currentPath === '/favorites') {
      history.push(`/favorites/${videoId}`);
    } else {
      history.push(`/video/${videoId}`);
    }
  }

  useEffect(() => {
    setState({
      ...state,
      videos,
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [videos]);

  const ExampleVideos = (props) => {
    return (
      <VideoContainer onClick={() => handleOnSelectVideo(props.video.videoId)}>
        <ThumbnailContainer duration={props.video ? props.video.duration : '00:15:30'}>
          <img src={props.video ? props.video.thumbnail : thumbnail} alt="Thumbnail" />
        </ThumbnailContainer>
        <InfoContainer>
          <div>
            <VideoName>{props.video ? props.video.title : 'videoTitle'}</VideoName>
            <ChannelName>
              {props.video ? props.video.channelTitle : 'Channel Name'}
            </ChannelName>
            <Description>
              {props.video ? props.video.description : 'description'}
            </Description>
          </div>
        </InfoContainer>
      </VideoContainer>
    );
  };

  return (
    <Container onTouchStart={touchStart} onTouchEnd={touchEnd}>
      <Title>{title}</Title>
      <Videos>
        {videoList.map((video, index) => (
          <ExampleVideos key={video?.videoId || index} index={index} video={video} />
        ))}
      </Videos>
    </Container>
  );
}

export default Main;
