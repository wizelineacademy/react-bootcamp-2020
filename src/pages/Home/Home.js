import React, { useContext , useEffect} from "react";
import VideoList from "../../components/VideoList/VideoList";
import PageContext from "../../providers/Context/PageContext";
import * as apiYoutube from "../../utils/apiYoutube";


const Home = () => {
  
  const { videoList , setVideoList } = useContext(PageContext);
  const { userLogged } = useContext(PageContext);

  const loadData = async () =>{
      if(videoList.length < 1){
        const varResponse = await apiYoutube.search();
        setVideoList(varResponse.data.items);
      }
  }
  useEffect(() => {
    loadData();
  });

  return (
    <>
      <h3>{userLogged.userStatus ? `Bienvenido ${userLogged.user}` : ""}</h3>
      <VideoList videos={videoList} spacing="-121" />
    </>
  );
};

export default Home;
