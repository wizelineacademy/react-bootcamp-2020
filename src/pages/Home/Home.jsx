import React, { useContext } from "react";
import { Link, Route, Redirect,Router } from "react-router-dom";
import ListVideos from "../../components/ListVideos/ListVideos";
import VideoContext from "../../providers/VideoContext";


const Home = () => {
  const {userSession } = useContext(VideoContext);
  
  return (
    <div>
      <Link to="/reproductorPlayer">
          <ListVideos />
      </Link>
      <Route render={() => userSession.loggedIn ? <Redirect to="/" /> : <Redirect to="/logIn" />} role="routerTest"/>
    </div>
  );
};
export default Home;
