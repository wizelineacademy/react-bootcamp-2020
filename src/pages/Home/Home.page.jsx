/* 
import React, { useRef } from 'react';
import { Link, useHistory } from 'react-router-dom';

import { useAuth } from '../../providers/Auth';
import './Home.styles.css';

function HomePage() {
  const history = useHistory();
  const sectionRef = useRef(null);
  const { authenticated, logout } = useAuth();

  function deAuthenticate(event) {
    event.preventDefault();
    logout();
    history.push('/');
  }

  return (
    <section className="homepage" ref={sectionRef}>
      <h1>Hello stranger!</h1>
      {authenticated ? (
        <>
          <h2>Good to have you back</h2>
          <span>
            <Link to="/" onClick={deAuthenticate}>
              ← logout
            </Link>
            <span className="separator" />
            <Link to="/secret">show me something cool →</Link>
          </span>
        </>
      ) : (
        <Link to="/login">let me in →</Link>
      )}
    </section>
  );
}

export default HomePage;
*/
import React, { useState } from 'react';
import Layout from '../../components/Layout';
import Logo from '../../components/Logo';
import SearchBar from '../../components/SearchBar';
import SuggestedSearches from '../../components/SuggestedSearches';

import './Home.styles.css';

function HomePage() {
  const [searchQuery, setSearchQuery] = useState('');

  return (
    <Layout>
      <Logo />
      <SearchBar initSearchQuery={searchQuery} onChange={setSearchQuery} isHomepage />
      <SuggestedSearches />
    </Layout>
  );
}

export default HomePage;
