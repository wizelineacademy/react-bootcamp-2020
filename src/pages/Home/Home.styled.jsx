import styled from 'styled-components';

export const HomeContainer = styled.div`
    width: 100%; 
    display: flex; 
    flex-Wrap: wrap;
    justify-content: center;
    background: ${({ color }) => color}
`;
