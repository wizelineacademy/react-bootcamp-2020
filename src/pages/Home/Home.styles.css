body {
  background-color: rgb(224, 219, 219);
}
