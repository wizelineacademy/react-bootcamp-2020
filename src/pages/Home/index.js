export { default } from './HomePage.component';
