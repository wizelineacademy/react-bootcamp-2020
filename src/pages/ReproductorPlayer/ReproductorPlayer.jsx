import React, { useContext } from "react";
import ListVideos from "../../components/ListVideos/ListVideos";
import VideoContainer from "../../components/VideoContainer/VideoContainer";
import { Route, Redirect } from "react-router-dom";
import VideoContext from "../../providers/VideoContext";
import "./ReproductorPlayer.style.css";
const ReproductorPlayer = () => {
  const { userSession,changingg } = useContext(VideoContext);
  
  return changingg ? (
    <div className="grid-container2">
      <VideoContainer />
      <ListVideos />
      <Route render={() =>  userSession.loggedIn ? <Redirect to='/reproductorPlayer' /> : <Redirect to='/logIn' />}
      />
    </div>
  ) : (
    <React.Fragment>
      <Route
        render={() =>
          userSession.loggedIn ? <Redirect to='/' /> : <Redirect to='/logIn' />
        }
      />
    </React.Fragment>
  );
};

export default ReproductorPlayer;
