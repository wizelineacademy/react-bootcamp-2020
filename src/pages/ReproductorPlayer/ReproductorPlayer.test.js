import React from 'react';
import {render } from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'
import VideoContext from "../../providers/VideoContext";
import ReproductorPlayer from './ReproductorPlayer';
import {BrowserRouter as Route} from "react-router-dom";
import VideoContainer from '../../components/VideoContainer/VideoContainer';
import ListVideos from "../../components/ListVideos/ListVideos";
import Home from "../Home/Home"

describe("<ReproductorPlayer/>",()=> {
    
    const mockValue = {
        setVideoMetaInfo: jest.fn(),
        changingg: false,
        changinggSetChangingg: jest.fn(),
        setFavoritesFlag: jest.fn(),
       
    }

    test("Rendering components within ReproductorPlayer Page",()=> {
        
        const videoProppertiesTest = {
            videoId : "id1",
            published : "SomeDate"
        }

        const userSessionTest ={
            loggedIn: true,
            user: "noSession"
        } 

        const changinggTest = true;

        const mockTrigger = {
            changingg : changinggTest,
            userSession : userSessionTest,
            videoPropperties : videoProppertiesTest
        }
        
        render(
        <Route>
            <VideoContext.Provider value={mockTrigger}>
                <ReproductorPlayer />
            </VideoContext.Provider>
        </Route>
        );
         
         expect(<VideoContainer/>).toBeTruthy();
         expect(<ListVideos/>).toBeTruthy();
         expect (<Home/>).toBeTruthy();
        
    })

    test("Redirect to logIn with changing false and loggedIn true",()=> {
        
        const videoProppertiesTest = {
            videoId : "id1",
            published : "SomeDate"
        }


        const userSessionTest ={
            loggedIn: false,
            user: "noSession"
        } ;
        const changinggTest = false;

        const mockTrigger = {
            changingg : changinggTest,
            userSession : userSessionTest,
            videoPropperties : videoProppertiesTest
        }
       
        const {container} = render(
        <Route>
            <VideoContext.Provider value={mockTrigger}>
                <ReproductorPlayer />
            </VideoContext.Provider>
        </Route>
        );
        container.querySelector('/logIn')
        expect(container.firstChild).toMatchSnapshot()
        
    })

    test("Redirect to Home with changing false and loggedIn true  ",()=> {
        
        const videoProppertiesTest = {
            videoId : "id1",
            published : "SomeDate"
        }

        const userSessionTest ={
            loggedIn: true,
            user: "noSession"
        }

        const changinggTest = false;

        const mockTrigger = {
            changingg : changinggTest,
            userSession : userSessionTest,
            videoPropperties : videoProppertiesTest
        }
       
        const {container} = render(
        <Route>
            <VideoContext.Provider value={mockTrigger}>
                <ReproductorPlayer />
            </VideoContext.Provider>
        </Route>
        );
       
        container.querySelector('/')
        expect(container.firstChild).toMatchSnapshot()
        
    })

})