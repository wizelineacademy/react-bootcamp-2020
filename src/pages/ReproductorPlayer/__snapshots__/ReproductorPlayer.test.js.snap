// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReproductorPlayer/> Redirect to Home with changing false and loggedIn true   1`] = `null`;

exports[`<ReproductorPlayer/> Redirect to logIn with changing false and loggedIn true 1`] = `null`;
